pkg/provisioner/cloudprovisioner.go

```diff
@@ -370,6 +370,7 @@ func (c *CloudProvisioner) PublishVolume(
 	attachResult = NewCloudAttachResult()
 	defer func() {
 		if !waitForCloud {
+			attachResult.ResultChannel() <- err
 			close(attachResult.ResultChannel())
```

Comment:
you are closing the channel twice - here and in the goroutine. This can result in a panic.